src/App.js

```diff
@@ -1,77 +1,71 @@
-import React from 'react';
+import {useState, useEffect} from 'react';
+import ReactGA from 'react-ga';
 import { div, h } from 'react-hyperscript-helpers';
 import Modal from 'react-modal';
 import './App.css';
 import DuosFooter from './components/DuosFooter';
 import DuosHeader from './components/DuosHeader';
+import {useHistory} from 'react-router-dom';
 
 import { SpinnerComponent as Spinner } from './components/SpinnerComponent';
 import { Storage } from './libs/storage';
 import Routes from './Routes';
 
+function App() {
+  const [loading] = useState(false);
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  let history = useHistory();
 
-class App extends React.Component {
-
-  constructor(props) {
-    super(props);
-    this.state = {
-      loading: false,
-      isLoggedIn: false
-    };
-  }
-
-  showSpinner = () => {
-    this.setState({
-      loading: true
+  const initializeReactGA = () => {
+    ReactGA.initialize('UA-173273990-1', {
+      titleCase: false
     });
   };
 
-  hideSpinner = () => {
-    this.setState({
-      loading: false
-    });
+  const trackPageView = (location) => {
+    ReactGA.pageview(location.pathname + location.search);
   };
 
-  async componentDidMount() {
-    const isLogged = await Storage.userIsLogged();
-    this.setState({ isLoggedIn: isLogged });
-  };
+  useEffect(() => {
+    const setUserIsLogged = async () => {
+      const isLogged = await Storage.userIsLogged();
+      setIsLoggedIn(isLogged);
+    };
 
-  signOut = () => {
+    Modal.setAppElement(document.getElementById('modal-root'));
+    initializeReactGA();
+    //call trackPageView to register initial page load
+    trackPageView(history.location);
+    //pass trackPageView as callback function for url change listener
+    history.listen(trackPageView);
+    setUserIsLogged();
+  }, []);
```

Comment:
This should fix the codacy warning
```suggestion
  }, [history]);
```

Reply:
Right, I was worried about potential re-renders on dependency updates which is why I opted for the empty array (only executes on mount), however the history instance doesn't change (only its properties) so passing it in as a dependency is the right call.